src/partition/repart.c

```diff
@@ -6113,6 +6289,177 @@ static int fd_apparent_size(int fd, uint64_t *ret) {
         return 0;
 }
 
+static bool need_fstab_one(const Partition *p) {
+        assert(p);
+
+        if (p->dropped)
+                return false;
+
+        if (!p->format)
+                return false;
+
+        if (p->n_mountpoints == 0)
+                return false;
+
+        return true;
+}
+
+static bool need_fstab(const Context *context) {
+        assert(context);
+
+        if (!arg_generate_fstab)
+                return false;
+
+        LIST_FOREACH(partitions, p, context->partitions)
+                if (need_fstab_one(p))
+                        return true;
+
+        return false;
+}
+
+static int context_fstab(Context *context) {
+        _cleanup_(unlink_and_freep) char *t = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *path = NULL;
+        int r;
+
+        assert(context);
+
+        if (!need_fstab(context)) {
+                log_notice("MountPoint= is not specified for any elligible partitions, not generating %s",
+                           arg_generate_fstab);
```

Comment:
this is now shown also when arg_generate_fstab is false? i.e. in the default case? that doesn't look. 

maybe just move the `if (!arg_generated_fstab)` check from need_fstab() into context_fstab()?